app/controllers/place_controller.rb

```diff
@@ -14,12 +12,15 @@ class PlaceController < ApplicationController
   REPORT_CHILD_ABUSE_SLUG = "report-child-abuse-to-local-council".freeze
 
   def show
+    set_content_item(PlacePresenter)
+
     if request.post?
       @location_error = location_error
       if @location_error
         @postcode = postcode
       elsif imminence_response.places_found?
-        @publication = PublicationWithPlacesPresenter.new(artefact, imminence_response.places)
+        @publication = PlacePresenter.new(content_item, imminence_response.places)
+        set_language_from_publication
```

Comment:
The code is confusing, as it actually is. `set_content_item` called at the top of this method calls `set_language_for_publication`. I believe its called again here as `set_language_from_publication` uses the value of `@publication` that has just been changed.

It could be moved below the end of the if, which might read better, but it would be called twice sometimes.